app/views/hackathon/subscriptions/unsubscribe_all.html.erb

```diff
@@ -0,0 +1,34 @@
+<% content_for :title, "Unsubscribe" %>
+
+<h1>Unsubscribe</h1>
+
+<% if @expired %>
+  <p>This link has expired.</p>
+<% else %>
+
+  <p>
+    Hey there, <b><%= @user.name || @user.email_address %></b>!
+  </p>
+
+  <p>
+    You have unsubscribed from all emails.
+    <% unless @unsubscribe_count.zero? %>
+      You were previous subscribed to
+      <%= @unsubscribe_count %> <%= "location".pluralize(@unsubscribe_count) %>.
+    <% end %>
+
+    <br/>
+    <i>Sad to see you go. :(</i>
+  </p>
+
+  <%# Button for undo-ing unsubscribe all %>
+  <% unless @unsubscribe_count.zero? %>
+    <%= form_with url: user_subscriptions_bulk_path, method: :put do |form| %>
+      <% @unsubscribed_ids.each do |id| %>
```

Comment:
```suggestion
    <% unless @subscriptions.none? %>
      You were previous subscribed to
      <%= @subscriptions.count %> <%= "location".pluralize(@subscriptions.count) %>.
    <% end %>

    <br/>
    <i>Sad to see you go. :(</i>
  </p>

  <%# Button for undo-ing unsubscribe all %>
  <% unless @subscriptions.none? %>
    <%= form_with url: user_subscriptions_bulk_path, method: :put do |form| %>
      <% @subscriptions.ids.each do |id| %>
```

Reply:
This unfortunately wouldn't work for a couple of reasons:
- `@subscriptions.ids` will be an empty array since there are no longer any active subscriptions. Therefore, we need to save the `id`s before unsubscribing. So they're held in `@unsubscribed_ids`.
- Same applies to `@subscriptions.count`. The count will also be zero since there are no longer any active subscriptions. `@unsubscribe_count` holds the number of Subscriptions that were unsubscribed by this request